### ISSUE TITLE
小模型训练到30epoch loss值不下降，infer视频之后发现SMPL body始终在人物上方

### ISSUE THREAD

**Issue by hitsz-zuoqi** [Arthur151/ROMP#177]
[V1_mbv3_h36m,mpiinf,coco,mpii,lsp,muco,crowdpose_g0,1,2,3.log](https://github.com/Arthur151/ROMP/files/8338065/V1_mbv3_h36m.mpiinf.coco.mpii.lsp.muco.crowdpose_g0.1.2.3.log)
可以帮忙看一下这个mobilenetv3的训练日志吗？
他的validation 指标下降到140&90附近就感觉卡住了，而且我把30epoch的模型拿出来测试了一下，发现SMPL body会飘在人物上面，我的想法是训练的时候body center heatmap应该是最先收敛的，所以即便pose上误差比较大，det loss应该收敛，也就是不会产生这种飘在人物上方的现象，不知道您训练的时候有没有这种问题，希望您能给出建议～

我在resnet的基础上将batch_size 调整为了128，lr 相应x2 为 0.0001
图示：
<img width="735" alt="image" src="https://user-images.githubusercontent.com/58206232/159833765-1131142f-e42d-4be8-bbda-e57d7b3f4bc9.png">

**Comment by Arthur151** [comment]
看log您这里训练没有使用预训练模型吧，通过其他GitHub developer的反馈，直接从0训ROMP比较难，很需要2D pose的预训练的，您可以像我这里使用[HigherHRNet-32](https://github.com/Arthur151/ROMP/releases/download/v1.1/pose_higher_hrnet_w32_512.pth)作为预训练模型，用其他2D pose训好的mobilenet开始也可以，但因为ROMP的图片处理方式不太一样，所以也要训一段时间，但应该不会陷入不收敛的情况。

还有一种简洁高效的思路，就是模型蒸馏，直接蒸大模型的输出就好了，我试过蒸小模型，很有效，很快就能训好。

关于偏移的问题，只是配置的问题，训练的时候设置的`smpl_mesh_root_align=True`,测试的时候也要这样设置，就好了。

**Comment by hitsz-zuoqi** [comment]
好的谢谢，我尝试之后有结论会分享出来的

**Comment by hitsz-zuoqi** [comment]
好的，关于蒸馏我直接用mse计算centermap以及paramsmap的teacher和student的loss，与gt给出的loss一起回传，请问我这样是否合理？
`        if self.distillation_learning:
            distillated_output = self.teacher_model(meta_data, **self.teacher_train_cfg)
            d_center_loss = 1000*self.distillation_loss(outputs['center_map'], distillated_output['center_map'])
            d_params_loss = 1000*self.distillation_loss(outputs['params_maps'],distillated_output['params_maps'])

        
        # for key,value in distillated_output.items():
            # print(key,value.shape)
        # print(outputs['center_map'].shape,outputs['params_maps'].shape)
        if not self.model_return_loss:
            outputs.update(self._calc_loss(outputs))
        loss, outputs = self.mutli_task_uncertainty_weighted_loss(outputs)
        # print(d_center_loss, d_params_loss, loss)
        loss = loss + d_center_loss + d_params_loss

        outputs['loss_dict']['d_center_loss'] = d_center_loss.item()
        outputs['loss_dict']['d_params_loss'] = d_params_loss.item()`

**Comment by Arthur151** [comment]
这段看起来没什么问题。可以这样先试试。

**Comment by hitsz-zuoqi** [comment]
https://user-images.githubusercontent.com/58206232/160743721-09eda6d9-6691-49ee-b4d0-cde57b8f8b74.mp4

您好，请问您知道这样子shape一会变大一会变小是怎么回事嘛？
这是我的video.yml配置
`
ARGS:
 tab: 'process_video'
 gpu: 0 

 backbone: 'mbv3'
 model_precision: 'fp32'
 val_batch_size: 8
 nw: 4
 model_path: /test/checkpoints/mbv3_cm64_V1_mbv3_on_gpu0,1,2,3_val/23_129.7699_84.7924_mbv3_cm64_V1_mbv3.pkl #  '../trained_models/ROMP_ResNet50_V1.pkl'
 smpl_mesh_root_align: True
 centermap_conf_thresh: 0.25

 make_tracking: True
 temporal_optimization: True
 smooth_coeff: 4.
 show_largest_person_only: True
 show_mesh_stand_on_image: False
 soi_camera: 'far' # 'close' # 'far'
 renderer: 'pytorch3d' # 'pyrender' 'pytorch3d' 

 inputs: 'demo/videos/sample_video6.mp4' # None
 output_dir: 'demo/sample_video6_results/'
 save_mesh: False
 save_centermap: False
 save_dict_results: True
 save_visualization_on_img: True
 fps_save: 24
 mesh_cloth: 'ghostwhite'  #'LightCyan'
`

**Comment by Arthur151** [comment]
我调一下，谢谢bug report。
建议直接用simple-romp，https://github.com/Arthur151/ROMP/blob/master/simple_romp/README.md
我后面主要维护simple-romp了，目前版本的代码将会弃用。

**Comment by Arthur151** [comment]
@hitsz-zuoqi ，原始输入视频可以发一下么？我用来测试debug。
其他视频也有这种问题么？

**Comment by hitsz-zuoqi** [comment]
原始视频在这：

https://user-images.githubusercontent.com/58206232/160774465-36ae0382-5cd3-491c-8e34-f33356aae625.mp4


pred_pose = quaternion.as_rotation_vector(quaternion.from_rotation_matrix(pred_pose_rotmat)).reshape(-1)
请问一下这一行里面的quaternion好像没有对应的包？coco14.py里面的函数

**Comment by Arthur151** [comment]
是的，单目深度的模糊性在半身的时候会更加明显，导致深度估计的不稳定，可以强化[translation的smooth coefficient]()https://github.com/Arthur151/ROMP/blob/8606d695e104f550a83bd74da3f3927c8e01faaf/romp/lib/utils/temporal_optimization.py#L19，来增强稳定性，强制稳定。
比如，{'cam': OneEuroFilter(1., .0).....}

**Comment by Arthur151** [comment]
@hitsz-zuoqi ,还是用simple-romp吧，simple-romp没有这个bug，其他人也向我汇报了这个bug了。

**Comment by hitsz-zuoqi** [comment]
好的，simple-romp是怎么解决这个bug的呢？

**Comment by Arthur151** [comment]
simple-romp去掉了不必要的操作，这样可能减少了bug的产生。
想固定trans参数，可以对trans估计求平均，然后将均值赋值trans，或者某一帧的结果也可以。。

**Comment by ArtiX-GP** [comment]
Hi, @hitsz-zuoqi 
did you manage to complete the distillation of the model?
